pkg/datapath/iptables/reconciler.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package iptables
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/stream"
+)
+
+type desiredState struct {
+	firstInit    bool
+	installRules bool
+	devices      []*tables.Device
+	localNode    node.LocalNode
+}
+
+func reconciliationLoop(
+	ctx context.Context,
+	health cell.HealthReporter,
+	installRules bool,
+	localNodeStore *node.LocalNodeStore,
+	db *statedb.DB,
+	devices statedb.Table[*tables.Device],
+	updateRules func(ctx context.Context, state *desiredState) error,
+) error {
+	state := desiredState{
+		firstInit:    true,
+		installRules: installRules,
+	}
+
+	localNode, err := localNodeStore.Get(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to get local node: %w", err)
+	}
+	state.localNode = localNode
+
+	localNodeEvents := stream.ToChannel(ctx, localNodeStore.Observable)
+out:
+	for {
+		txn := db.ReadTxn()
+		var devicesWatch <-chan struct{}
+		state.devices, devicesWatch = tables.SelectedDevices(devices, txn)
+
```

Comment:
Might be worth adding rate-limiting here to do this e.g. max once per second (e.g. `rate.NewLimiter(time.Second, 1)`)